### PR TITLE
New regression tests for self hosted "login" (sites fetching)

### PR DESCRIPTION
**Pull request by maxme**
`testSelfHostedHTTPAuthFetchSites` and `testSelfHostedSelfSignedSSLFetchSites` are just stubs (FAILING tests) - HTTP Auth and SSL warning are not implemented yet.

You need to fill `example/tests.properties` file with real credentials, I'll push an encrypted version once we have something solid - ping me and I'll send you my current version.

Note: Also added a debug button and a LogUtils method for debugging.

### CODE REVIEW

File: example/tests.properties-example [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+TEST_WPCOM_USERNAME_TEST1 = FIXME
+TEST_WPCOM_PASSWORD_TEST1 = FIXME
+
+# Simple Self hosted
+TEST_WPORG_USERNAME_SH_SIMPLE = FIXME
+TEST_WPORG_PASSWORD_SH_SIMPLE = FIXME
+TEST_WPORG_URL_SH_SIMPLE = FIXME
+
+# Simple Self hosted HTTP - contributor role
+TEST_WPORG_USERNAME_SH_SIMPLE = FIXME
```
Comment (aforcier) [inline example/tests.properties-example]: These three fields should end in `_CONTRIB` (it's correct in the test.)

File: example/tests.properties-example [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Test WPCOM user
+TEST_WPCOM_USERNAME_TEST1 = FIXME
+TEST_WPCOM_PASSWORD_TEST1 = FIXME
+
+# Simple Self hosted
+TEST_WPORG_USERNAME_SH_SIMPLE = FIXME
+TEST_WPORG_PASSWORD_SH_SIMPLE = FIXME
+TEST_WPORG_URL_SH_SIMPLE = FIXME
+
```
Comment (aforcier) [inline example/tests.properties-example]: These three fields should end in `_CONTRIB` (it's correct in the test.)

File: example/tests.properties-example [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+# Simple Self hosted HTTP - contributor role
+TEST_WPORG_USERNAME_SH_SIMPLE = FIXME
+TEST_WPORG_PASSWORD_SH_SIMPLE = FIXME
+TEST_WPORG_URL_SH_SIMPLE = FIXME
+
+# Multi site Self Hosted
+TEST_WPORG_USERNAME_SH_MULTISITE = FIXME
+TEST_WPORG_PASSWORD_SH_MULTISITE = FIXME
+TEST_WPORG_URL_SH_MULTISITE = FIXME
+
```
Comment (aforcier) [inline example/tests.properties-example]: HTTP self-hosted and self-signed SSL are missing an `_SH_` (correct in the tests).

**Comment by aforcier** [comment]
We can drop all the test login info from `gradle.properties-example` now:
https://github.com/wordpress-mobile/WordPress-Stores-Android/blob/31f3933d371b6d8ae581af6b82b24813c5ae532b/example/gradle.properties-example

**Comment by maxme** [comment]
Ah right, I forgot to update examples.

**Comment by aforcier** [comment]
Other than the things I mentioned, looks good! Tests that should be passing are passing, and I checked that the volley cache fix is still working on the official version we're switching to in this PR.

**Comment by maxme** [comment]
Example files updated in  3da9bbd

**Comment by aforcier** [comment]
:shipit:

**Pull request merged**